samples/oauth_rails_vethealth/app/models/health_api_response.rb

```diff
@@ -0,0 +1,55 @@
+class HealthApiResponse
+  attr_reader :api_name, :target, :page, :count
+  def initialize(api_name:, id:, access_token:, action: :read, page: 1, count: 10)
+    @api_name, @id, @action = api_name, id, action
+
+    case action
+    when :search
+      @page, @count = page, count
+      @target = "https://dev-api.va.gov/services/argonaut/v0/#{api_name}?#{search_param_name}=#{id}&page=#{page}&_count=#{count}"
+    when :read
+      @page, @count = nil, nil
+      @target = "https://dev-api.va.gov/services/argonaut/v0/#{api_name}/#{id}"
+    else
+      raise "action must be one of the symbols :search or :read"
+    end
+
+    @api_response = HTTParty.get(@target, headers: { Authorization: "Bearer #{access_token}" })
+  end
+
+  def code
+    @api_response.code
+  end
+
+  def response_string
+    # use JSON pretty_generate to add readable whitespace and gsub api links to be links with the test app
+    @response_string ||= JSON.pretty_generate(@api_response.to_h).gsub(/\"(https:\/\/dev-api.va.gov\/services\/argonaut\/v0)\/(\w+)\/(.+)\"/, "<a href=\"/health_api/api_response/\\2/\\3\">\\1\/\\2\/\\3</a>")
+  end
+
+  def api_navs
+    return @api_navs if @api_navs
+    return nil unless (@action == :search && code == 200)
+    @api_navs = []
+    self_nav = nil
+    @api_response['link'].each do |link|
```

Comment:
I think you could probably memoize `@api_navs` if you moved this logic into Yet Another Method, but I don't have strong feelings about it

Reply:
i believe it is memoized by returning @api_navs a couple lines up

Reply:
Oh yeah, I just meant in the sense of using the handy ruby `||=` operator, but it doesn't make a difference really